src/main/kotlin/com/exactpro/th2/rptdataprovider/handlers/messages/MessageExtractor.kt

```diff
@@ -66,10 +68,9 @@ class MessageExtractor<B, G, RM, PM>(
     private var lastElement: StoredMessageId? = null
     private var lastTimestamp: Instant? = null
 
-    private val order = if (request.searchDirection == TimeRelation.AFTER) {
-        Order.DIRECT
-    } else {
-        Order.REVERSE
+    private val order = when(request.searchDirection) {

```

Comment:
note: we probably should add an explicit type for `order` property - right now it is a bit hard to understand what type it has